sentry-ruby/lib/sentry/net/http.rb

```diff
@@ -49,10 +49,12 @@ def request(req, body = nil, &block)
 
       private
 
-      def set_sentry_trace_header(req, sentry_span)
+      def set_sentry_trace_header(req, sentry_span, request_info)
         return unless sentry_span
 
         client = Sentry.get_current_client
+        return unless client
```

Comment:
Is there a case where Span is initialised (which means the SDK is initialised too?) and but client is `nil`?

Reply:
only if there's some weird multi-threaded race, since we had a couple of bugs like that before in redis, just wanted to be safe

Reply:
If you mean #2057, it's addressed with SDK initialisation check.

And in our case here, `set_sentry_trace_header` is only called in `HTTP#request`, which already checked `Sentry.initialized?`. `Sentry.with_child_span` also yields `nil` if the SDK is not initialised, which we also handled at the beginning of this method.

Given we already have 2 checks on the same thing, I feel we're covered from the issue you mentioned. And to me this actually feels a bit misleading instead.

Reply:
removed